### PR TITLE
Support extensions with multiple parts for gitlab (i.e. en.md)

### LINKED ISSUE #1462: [Bug] Cannot list multi-part extension files
**- Do you want to request a *feature* or report a *bug*?**
Bug

**- What is the current behavior?**
You can't list files using a multi-part extension.

**- If the current behavior is a bug, please provide the steps to reproduce.**
- Forked the one-click-hugo-cms: https://gitlab.com/Nic128/one-click-hugo-cms/
- Rename the posts from `.md` to `.en.md`
- Set the extension for the posts and format to:
```
extension: 'en.md'
format: 'frontmatter'
```
- Create a new post, the post will be created properly. `{{slug}}.en.md`
- You can't see any file listed for editing, even if the files exist.
- I see the list data from ajax calls, but the filtering in the cms code is filtering everything out.

**- What is the expected behavior?**
If just using `md` list the files correctly, `en.md` should list the same files.

I found out how to use multi-part extension from this thread. https://github.com/netlify/netlify-cms/issues/1399

**- Please mention your versions where applicable.**

Netlify CMS version: 1.9.2
Browser version: Firefox 60

Node.JS version: 8.11.2
Operating System: OSX

**- Please link or paste your `config.yml` below if applicable.**
https://gitlab.com/Nic128/one-click-hugo-cms/blob/master/site/static/admin/config.yml

### PR DESCRIPTION
**Pull request by Nic128**
**- Summary**
Fixes #1462. See discussion there for context.

**- Test plan**
I ported the exact fix from #1123  to gitlab.

**- Description for the changelog**
Support extensions with multiple parts for gitlab (i.e. en.md)

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: src/backends/gitlab/implementation.js [hunk 1]
```diff
@@ -66,7 +66,9 @@ export default class GitLab {
   entriesByFolder(collection, extension) {
     return this.api.listFiles(collection.get("folder"))
     .then(({ files, cursor }) =>
-      this.fetchFiles(files.filter(file => fileExtension(file.name) === extension))
+      this.fetchFiles(files.filter(files =>
+        files.filter(file => file.name.endsWith('.' + extension))
+      )
```
Comment (tech4him1) [inline src/backends/gitlab/implementation.js]: Bit of a typo here -- this whole block should be changed to:

```js
this.fetchFiles(files.filter(file => file.name.endsWith('.' + extension)))
```

**Comment by verythorough** [comment]
Deploy preview for *cms-demo* ready!

Built with commit a33cb8b272f4e56466d041563b842f3b6cf28fd8

https://deploy-preview-1478--cms-demo.netlify.com

**Comment by Nic128** [comment]
Changes done.

**Comment by tech4him1** [comment]
@Nic128 I'll try to get this merged later today or tomorrow

**Comment by Nic128** [comment]
There are more places that use the `fileExtension` helper. I'd rather not remove it in this PR.

**Comment by Nic128** [comment]
Done.

**Comment by Nic128** [comment]
Is there more to do for this PR?

**Comment by tech4him1** [comment]
So sorry, I've been pretty much offline for the last 2 weeks. I'll try to review ASAP.

@erquhart Maybe we can get this in the next patch release if at all possible?

**Comment by tech4him1** [comment]
Released in v1.9.4.

**Comment by tech4him1** [comment]
@Nic128 Released in v1.9.4.

**Pull request merged**